src/enum_net.cpp

```diff
@@ -200,41 +205,34 @@ namespace {
 
 			if (nl_hdr->nlmsg_type == NLMSG_DONE) break;
 
-			msg_len += read_len;
+			read_cursor += read_len;
 
 			if ((nl_hdr->nlmsg_flags & NLM_F_MULTI) == 0) break;
 		}
-		return msg_len;
+		buf.resize(read_cursor);
+		return 0;
 	}
 
-	constexpr int NL_BUFSIZE = 8192;
-
-	int nl_dump_request(int sock, std::uint16_t type, std::uint32_t seq, char family, span<char> msg, std::size_t msg_len)
+	int nl_dump_request(int const sock, std::uint16_t const type
+		, std::uint32_t const seq, std::vector<char>& msg
+		, nlmsghdr* const request_msg)
 	{
-		nlmsghdr* nl_msg = reinterpret_cast<nlmsghdr*>(msg.data());
-		nl_msg->nlmsg_len = std::uint32_t(NLMSG_LENGTH(msg_len));
-		nl_msg->nlmsg_type = type;
-		nl_msg->nlmsg_flags = NLM_F_DUMP | NLM_F_REQUEST;
-		nl_msg->nlmsg_seq = seq;
+		request_msg->nlmsg_type = type;
+		request_msg->nlmsg_flags = NLM_F_DUMP | NLM_F_REQUEST;
+		request_msg->nlmsg_seq = seq;
 		// in theory nlmsg_pid should be set to the netlink port ID (NOT the process ID)
 		// of the sender, but the kernel ignores this field so it is typically set to
 		// zero
-		nl_msg->nlmsg_pid = 0;
-		// first byte of routing messages is always the family
-		msg[sizeof(nlmsghdr)] = family;
```

Comment:
I wanted to get rid of this line (and assumption). Instead, I pass in the netlink request message, and it's allocated on the stack by the caller. I don't think there's a good reason to reuse the buffer that we read the response into for the request.